pkg/services/datasources/service/secrets_mig.go

```diff
@@ -0,0 +1,52 @@
+package service
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/services/datasources"
+)
+
+type DataSourceSecretMigrationService struct {
+	dataSourcesService datasources.DataSourceService
+}
+
+func ProvideDataSourceMigrationService(dataSourcesService datasources.DataSourceService) *DataSourceSecretMigrationService {
+	return &DataSourceSecretMigrationService{
+		dataSourcesService: dataSourcesService,
+	}
+}
+
+func (s *DataSourceSecretMigrationService) Migrate(ctx context.Context) error {
```

Comment:
Do you have an idea of how long this till take to run, considering sql secret store vs plugin secrets store and feature toggle enabled/disabled? Depending on the answer or as a general recommendation you might want to document this in the upgrade notes, similar to https://grafana.com/docs/grafana/latest/setup-grafana/upgrade-grafana/#annotations-database-migration

Reply:
Created a draft task for this on our board and will followup closer to release.

https://github.com/orgs/grafana/projects/87/views/12